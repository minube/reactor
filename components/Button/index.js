export { default } from './Button'; // eslint-disable-line
