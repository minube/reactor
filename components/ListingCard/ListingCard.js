import {
  func, node, shape, string,
} from 'prop-types';
import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';

import { LAYOUT, SHAPE } from '../../common';
import { PictureCard, Rating, Text } from '..';
import styles from './ListingCard.style';

const ListingCard = ({
  category, children, description, onPress, rating = {}, title, ...inherit
}) => (
  <View style={inherit.style}>
    <TouchableWithoutFeedback disabled={!onPress} onPress={onPress}>
      <View style={[styles.container, LAYOUT.STYLE.CARD.WIDTH]}>
        <PictureCard {...inherit} style={[]} />
        <View style={styles.content}>
          { category && (
          <Text lighten tiny>
            {category}
          </Text>
          ) }
          { title && (
          <Text bold small numberOfLines={1} ellipsizeMode="tail" style={styles.title}>
            {title}
          </Text>
          ) }
          { rating.value && <Rating {...rating} style={styles.rating} /> }
          { description && (
          <Text lighten tiny numberOfLines={3} ellipsizeMode="tail">
            {description}
          </Text>
          ) }
          { children }
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
);

ListingCard.propTypes = {
  category: string,
  children: node,
  description: string,
  image: string,
  onPress: func,
  rating: shape(SHAPE.RATING),
  title: string,
};

ListingCard.defaultProps = {
  category: undefined,
  children: undefined,
  description: undefined,
  image: undefined,
  onPress: undefined,
  rating: {},
  title: undefined,
};

export default ListingCard;
