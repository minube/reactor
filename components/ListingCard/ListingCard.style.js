import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  COLOR, FONT, LAYOUT, OFFSET, UNIT, STYLE,
} = THEME;

export default StyleSheet.create({
  container: {
    margin: OFFSET / 2,
    width: LAYOUT.CARD_MAX_WIDTH,
    overflow: 'hidden',
  },

  empty: {
    borderColor: COLOR.BORDER,
    borderWidth: 1,
  },

  elevation: {
    ...STYLE.SHADOW,
  },

  content: {
    padding: UNIT,
  },

  rating: {
    width: FONT.SIZE.SMALL,
    height: FONT.SIZE.SMALL,
    tintColor: COLOR.PRIMARY,
  },

  title: {
    fontSize: FONT.SIZE.REGULAR,
    marginBottom: UNIT / 2,
  },

  description: {
    fontSize: FONT.SIZE.TINY,
  },

  commentContent: {
    alignItems: 'flex-start',
    marginTop: UNIT,
  },

  avatar: {
    width: LAYOUT.AVATAR_SMALL,
    height: LAYOUT.AVATAR_SMALL,
    borderRadius: LAYOUT.AVATAR_SMALL / 2,
    marginRight: UNIT,
    backgroundColor: COLOR.BACKGROUND,
  },

  comment: {
    fontSize: FONT.SIZE.TINY,
    opacity: 0.35,
    maxHeight: FONT.SIZE.TINY * 3.6,
    overflow: 'hidden',
    width: '87.5%',
  },
});
