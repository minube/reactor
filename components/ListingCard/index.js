export { default } from './ListingCard'; // eslint-disable-line
