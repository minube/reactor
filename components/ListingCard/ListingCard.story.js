import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, number } from '@storybook/addon-knobs/react';

import { THEME, STYLE } from '../../common';
import Button from '../Button';
import ListingCard from './ListingCard';
import Price from '../Price';
import Text from '../Text';

const { COLOR: { TEXT, WARNING }, FONT, UNIT } = THEME;
const LIPSUM_IMAGE = 'https://picsum.photos/320/200/?random';

storiesOf('ListingCard', module)
  .addWithJSX('Empty', () => (
    <ListingCard empty />
  ))
  .addWithJSX('default', () => (
    <ListingCard
      category="Restaurantes"
      image={LIPSUM_IMAGE}
      rating={4}
      ratingCount={2}
      title="Café Les Deux Molins - El café de Amelie con titulo súper largo puede..."
    />
  ))
  .addWithJSX('with description', () => (
    <ListingCard
      category="Restaurantes"
      description="Es una de esas joyas de Río que no hay que perderse. Se encuentra a caballo entre Lapa y el Barrio de… "
      image={LIPSUM_IMAGE}
      rating={4}
      title="Café Les Deux Molins - El café de Amelie con titulo súper largo la"
    />
  ))
  .addWithJSX('with button as a child', () => (
    <ListingCard
      category="Monumentos"
      image={LIPSUM_IMAGE}
      rating={4}
      ratingCount={2}
      title="Torre Eiffel Lorem ipsum dolor sit amet adipiscing "
    >
      <Button flat rounded small title="15 Actividades" style={{ alignSelf: 'flex-start', marginHorizontal: 0 }} />
    </ListingCard>
  ))
  .addWithJSX('with a container as a child', () => (
    <ListingCard
      category="Monumentos"
      image={LIPSUM_IMAGE}
      rating={4}
      ratingCount={2}
      title="Torre Eiffel Lorem ipsum dolor sit amet adipiscing "
    >
      <View style={STYLE.ROW}>
        <View style={{ marginRight: UNIT }}>
          <Text style={{ color: TEXT, fontSize: FONT.SIZE.TINY }}>desde</Text>
          <Price value={90} symbol="$" style={{ fontSize: FONT.SIZE.LARGE }} />
        </View>
        <Text style={{
          fontSize: FONT.SIZE.TINY, fontWeight: FONT.WEIGHT.BOLD, color: WARNING, width: '50%',
        }}>
          Mascotas bienvenidas
        </Text>
      </View>
    </ListingCard>
  ))
  .addWithJSX('💄 Custom Style', () => (
    <ListingCard />
  ))
  .addWithJSX('🏀 Playground', () => (
    <ListingCard
      category={text('category', 'category')}
      description={text('description', 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s,')}
      image={text('image', LIPSUM_IMAGE)}
      onPress={action('Button.onPress()')}
      rating={number('rating', 4)}
      ratingCount={number('ratingCount', 2)}
      title={text('title', 'title')}
    />
  ));

