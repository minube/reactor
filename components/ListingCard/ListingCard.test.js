import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Button from '../Button';
import ListingCard from './ListingCard';

const { COLOR, SPACE } = THEME;
const LIPSUM_IMAGE = 'https://picsum.photos/320/200/?random';
const RATING = { showValue: false, count: 12, value: 4.5 };

describe('<ListingCard>', () => {
  it('renders', () => {
    const tree = renderer.create(<ListingCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {image}', () => {
    const tree = renderer.create(<ListingCard image={LIPSUM_IMAGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<ListingCard title="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {category}', () => {
    const tree = renderer.create(<ListingCard category="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {description}', () => {
    const tree = renderer.create(<ListingCard description="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {rating}', () => {
    const tree = renderer.create(<ListingCard rating={RATING} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(
      <ListingCard>
        <Button flat rounded small title="15 Actividades" style={{ alignSelf: 'flex-start', marginHorizontal: 0 }} />
      </ListingCard>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {styleContainer}', () => {
    const tree = renderer.create(<ListingCard styleContainer={{ backgroundColor: COLOR.PRIMARY, paddingTop: SPACE.XS }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
