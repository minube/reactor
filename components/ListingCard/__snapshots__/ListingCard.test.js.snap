// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListingCard> renders 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`<ListingCard> when {category} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
          }
        }
      >
        Hello world
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ListingCard> when {children} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 22.5,
            "margin": 5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#ffffff",
              "borderColor": "#dbdbdb",
              "borderRadius": 22.5,
              "borderWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 30,
              "justifyContent": "center",
              "marginHorizontal": 0,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={undefined}
            style={
              Object {
                "color": "#9b9b9b",
                "fontFamily": undefined,
                "fontSize": 12,
                "fontWeight": "600",
                "lineHeight": 20.8,
              }
            }
          >
            15 Actividades
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ListingCard> when {description} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": undefined,
            "fontSize": 12,
            "lineHeight": 15.600000000000001,
            "maxHeight": 46.800000000000004,
            "overflow": "hidden",
          }
        }
      >
        Hello world
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ListingCard> when {image} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="cover"
            source={
              Object {
                "uri": "https://picsum.photos/320/200/?random",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`<ListingCard> when {rating} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#10c0bc",
              "fontFamily": undefined,
              "fontSize": 16,
              "lineHeight": 16,
              "marginBottom": 5,
              "marginRight": 2.5,
            }
          }
        >
          ★
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#10c0bc",
              "fontFamily": undefined,
              "fontSize": 16,
              "lineHeight": 16,
              "marginBottom": 5,
              "marginRight": 2.5,
            }
          }
        >
          ★
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#10c0bc",
              "fontFamily": undefined,
              "fontSize": 16,
              "lineHeight": 16,
              "marginBottom": 5,
              "marginRight": 2.5,
            }
          }
        >
          ★
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#10c0bc",
              "fontFamily": undefined,
              "fontSize": 16,
              "lineHeight": 16,
              "marginBottom": 5,
              "marginRight": 2.5,
            }
          }
        >
          ★
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#10c0bc",
              "fontFamily": undefined,
              "fontSize": 16,
              "lineHeight": 16,
              "marginBottom": 5,
              "marginRight": 2.5,
            }
          }
        >
          ☆
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Object {
              "color": "#9b9b9b",
              "fontFamily": undefined,
              "fontSize": 12,
              "lineHeight": 15.600000000000001,
              "marginBottom": 5,
            }
          }
        >
           (12)
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ListingCard> when {style} 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#10c0bc",
      "overflow": "hidden",
      "paddingTop": 10,
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`<ListingCard> when {title} 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderRadius": 3,
              "height": 190,
              "justifyContent": "center",
              "opacity": 0.8,
              "width": 168,
            }
          }
        >
          <Image
            onLoad={[Function]}
            resizeMode="center"
            source={
              Object {
                "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
              }
            }
            style={
              Object {
                "borderRadius": 3,
                "height": 190,
                "opacity": 0.8,
                "width": 168,
              }
            }
          />
          <ActivityIndicator
            animating={true}
            color="white"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "position": "absolute",
              }
            }
          />
        </View>
        <View
          pointerEvents="none"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 18.2,
            "marginVertical": 5,
          }
        }
      >
        Hello world
      </Text>
    </View>
  </View>
</View>
`;
